packages/resource-deployment/scripts/pool-startup/pool-startup.sh

```diff
@@ -4,26 +4,61 @@
 # Licensed under the MIT License.
 
 # shellcheck disable=SC1090
+set -eo pipefail
 
-# intentionally ignored -e option for this file, to not interrupt retry logic
-set -o pipefail
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -k <key vault name>
+"
+    exit 1
+}
+
+waitForApplicationUpdates() {
+    echo "Waiting for other application updates"
+    while fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock* >/dev/null 2>&1; do
+        echo "waiting..."
+        sleep 5
+    done
+}
 
-runWithRetry() {
-    local retryCount=0
-    local maxRetryCount=5
+installBootstrapPackages() {
+    # wait for OS updates on reboot
+    waitForApplicationUpdates
 
-    until "${0%/*}/pool-startup-internal.sh"; do
-        ((retryCount++))
+    echo "Restoring dpkg configuration"
+    dpkg --configure -a
+    waitForApplicationUpdates
 
-        if ((retryCount == maxRetryCount)); then
-            echo "Maximum retry count reached. Pool startup script failed"
-            exit 1
-        fi
-        echo "Retry count - $retryCount. Pool startup script failed"
-    done
+    echo "Running apt-get update"
+    apt-get update
+    waitForApplicationUpdates
+
+    echo "Installing curl"
+    apt-get install -y curl
+    waitForApplicationUpdates
 
-    echo "Successfully completed pool startup script execution after retry count - $retryCount"
-    exit 0
+    echo "Installing az cli"
+    curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+    waitForApplicationUpdates
 }
 
-runWithRetry
+# Read script arguments
+while getopts "k:" option; do
+    case $option in
+    k) KEY_VAULT_NAME=${OPTARG} ;;
+    *) exitWithUsageInfo ;;
+    esac
+done
+
+if [[ -z $KEY_VAULT_NAME ]]; then
+    exitWithUsageInfo
+fi
+
+installBootstrapPackages
+
+"${0%/*}/pull-image-from-container-registry.sh"
+
+echo "Invoking custom pool startup script"
+"${0%/*}/custom-pool-post-startup.sh"
```

Comment:
Why are we no longer retrying here?

Reply:
We are retrying using Batch settings for the startup script (part of this pr)